mtags/src/main/scala-2/scala/meta/internal/pc/ScalaPresentationCompiler.scala

```diff
@@ -179,14 +180,18 @@ case class ScalaPresentationCompiler(
   override def convertToNamedArguments(
       params: OffsetParams,
       argIndices: ju.List[Integer]
-  ): CompletableFuture[ju.List[TextEdit]] = {
-    val empty: ju.List[TextEdit] = new ju.ArrayList[TextEdit]()
+  ): CompletableFuture[jm.Either[String, ju.List[TextEdit]]] = {
+    val empty: jm.Either[String, ju.List[TextEdit]] =
+      jm.Either.forRight(new ju.ArrayList[TextEdit]())
     compilerAccess.withInterruptableCompiler(empty, params.token) { pc =>
       new ConvertToNamedArgumentsProvider(
         pc.compiler(),
         params,
         argIndices.asScala.map(_.toInt).toSet
-      ).convertToNamedArguments.asJava
+      ).convertToNamedArguments match {
+        case Left(error) => jm.Either.forLeft(error)
+        case Right(edits) => jm.Either.forRight(edits.asJava)
```

Comment:
Not pretty, but it will be packed in the future.
```suggestion
        case Left(error) => throw new Exception(error)
        case Right(edits) =>  edits
```

Reply:
Just doing that results in `withInterruptableCompiler` returning default and shutting down the compiler (I think). Basically `CompilerAccess` has some ways to deal with errors and recovering from them. Passing a special subtype of exceptions around to allow them to fall through is probably a possibility but I'm not sure if it's worth it or a cleaner solution. What do you think?

Reply:
Maybe we could allow returning a CompletableFuture instead of throwing? And it could be CompletableFuture.failed (or something along these lines) and CompletableFuture.success

This would mean chaning the signature though, so might also be more work.

Reply:
Or you could map a the end?

```
 val res = compilerAccess.withInterruptableCompiler(empty, params.token) { pc =>
      new ConvertToNamedArgumentsProvider(
        pc.compiler(),
        params,
        argIndices.asScala.map(_.toInt).toSet
      ).convertToNamedArguments.asJava
    }
res.thenComposeAsync{
      case Left(error) => CompletableFuture.failedFuture[ju.List[TextEdit]](error)
      case Right(result) => CompletableFuture.completedFuture(result)
    }
```

Reply:
It seems perfectly simple and I did that but it has to be `case Left(error: String) => ...` and raises a warning, because of erasure.